meteor/client/ui/SegmentTimeline/Renderers/CustomLayerItemRenderer.tsx

```diff
@@ -182,6 +200,23 @@ export class CustomLayerItemRenderer<
 		) : null
 	}
 
+	renderDuration() {
+		const uiPiece = this.props.piece
+		const innerPiece = uiPiece.instance.piece
+		const content = innerPiece.content
+		const duration = content && content.sourceDuration
+		if (duration && this.props.showDurationSourceLayers?.has(innerPiece.sourceLayerId)) {
```

Comment:
Why pass an entire set down to the `CustomLayerItemRender`? Shouldn't just the renderer be informed that it either should or shouldn't render the duration? It has no reason to know what other sourceLayers should have this enabled. This should just be a simple boolean at this stage.